engine/src/state_chain_observer/client/extrinsic_api/signed/submission_watcher.rs

```diff
@@ -282,6 +283,13 @@ impl<'a, 'env, BaseRpcClient: base_rpc_api::BaseRpcApi + Send + Sync + 'static>
 
 							self.runtime_version = new_runtime_version;
 						},
+						ClientError::Call(obj)
+							if obj.code() == 1002 &&
+								obj.message().contains("State already discarded") =>
```

Comment:
makes it a little less likely this breaks later with updates
```suggestion
								obj.message()
								    .to_lowercase()
						            .contains("state already discarded") =>
```